.travis.yml

```diff
@@ -16,53 +24,134 @@ addons:
+        # Cython code, you can set CONDA_DEPENDENCIES=''
+        - CONDA_DEPENDENCIES='Cython sherpa'
+        - CONDA_DEPENDENCIES_DOC='Cython sphinx-astropy sherpa'
+        
+        # List other runtime dependencies for the package that are available as
+        # pip packages here.
+        - PIP_DEPENDENCIES=''
+
+        # Conda packages for affiliated packages are hosted in channel
+        # "astropy" while builds for astropy LTS with recent numpy versions
+        # are in astropy-ci-extras. If your package uses either of these,
+        # add the channels to CONDA_CHANNELS along with any other channels
+        # you want to use.
+        - CONDA_CHANNELS='astropy-ci-extras astropy sherpa'
+
+        # If there are matplotlib or other GUI tests, uncomment the following
+        # line to use the X virtual framebuffer.
+        # - SETUP_XVFB=True
+
+        # If you want to ignore certain flake8 errors, you can list them
+        # in FLAKE8_OPT, for example:
+        # - FLAKE8_OPT='--ignore=E501'
+        - FLAKE8_OPT=''
 
     matrix:
-        - SETUP_CMD='egg_info'
+        # Make sure that egg_info works without dependencies
+        - PYTHON_VERSION=3.7 SETUP_CMD='egg_info'
+            - gfortran
+
 
 matrix:
+
+    # Don't wait for allowed failures
+    fast_finish: true
+
     include:
+        # Try MacOS X
+        - os: osx
+          env: SETUP_CMD='test'
 
-        # Do a coverage test in Python 2.
-        - python: 2.7
+        # Do a coverage test.
+        - os: linux
           env: SETUP_CMD='test --coverage'
 
-        # Sherpa 4.8.2 is the first to support python
-        - python: 3.5
-          env: SHERPA_VERSION>=4.8.2 
+        # Check for sphinx doc build warnings - we do this first because it
+        # may run for a long time
+        - os: linux
+          env: SETUP_CMD='build_docs -w'
+               CONDA_DEPENDENCIES=$CONDA_DEPENDENCIES_DOC
+
+        # Now try Astropy dev with the latest Python and LTS with Python 2.7 and 3.x.
+        - os: linux
+          env: ASTROPY_VERSION=development
+               EVENT_TYPE='pull_request push cron'
+        - os: linux
+          env: ASTROPY_VERSION=lts
+
+        # Try all python versions and Numpy versions. Since we can assume that
+        # the Numpy developers have taken care of testing Numpy with different
+        # versions of Python, we can vary Python and Numpy versions at the same
+        # time.
 
-        - python: 3.5
-          env: SHERPA_VERSION>=4.8.2 ASTROPY_VERSION=dev
+        - os: linux
+          env: PYTHON_VERSION=3.5 NUMPY_VERSION=1.12
+        - os: linux
+          env: PYTHON_VERSION=3.6 NUMPY_VERSION=1.13
+        - os: linux
+          env: NUMPY_VERSION=1.14
 
-        # Testing the docs building
-        - python: 3.5
-          env: SHERPA_VERSION>=4.8.2 SETUP_CMD='build_sphinx -w'
-               CONDA_DEPENDENCIES='sherpa ipython'
+        # Try numpy pre-release
```

Comment:
given that `saba` is a low traffic package I would suggest to test against the numpy dev version rather than the prerelease, given that the pre-release period is not that long.

Reply:
(and we should probably change this recommendation in the package-template, too)